### PR TITLE
Remove managed, provided dependency on log4j 1.2.17

### PR DESCRIPTION
**Pull request by basil**
`bom/pom.xml` ships with a provided dependency on log4j 1.2.17 in its `<dependencyManagement>` section. This puts the actual log4j 1.2.17 on the compile and test classpath for core and plugins, but not the runtime classpath. On the runtime classpath, the same classes are delivered by `log4j-over-slf4j`.

The cost of this is (a) having the real (vulnerable!) log4j 1.2.17 JAR on our compile and test classpaths and (b) false positive vulnerability alerts for the managed, provided dependency on log4j 1.2.17. What are the benefits?

To find out, I added Zookeeper 3.6.1 as a dependency to a plugin. Zookeeper 3.6.1 depends on log4j 1.2.17:

```
[INFO] \- org.apache.zookeeper:zookeeper:jar:3.6.1:compile
[INFO]    +- commons-lang:commons-lang:jar:2.6:compile
[INFO]    +- org.apache.zookeeper:zookeeper-jute:jar:3.6.1:compile
[INFO]    +- org.apache.yetus:audience-annotations:jar:0.5.0:compile
[INFO]    +- io.netty:netty-handler:jar:4.1.48.Final:compile
[INFO]    |  +- io.netty:netty-common:jar:4.1.48.Final:compile
[INFO]    |  +- io.netty:netty-resolver:jar:4.1.48.Final:compile
[INFO]    |  +- io.netty:netty-buffer:jar:4.1.48.Final:compile
[INFO]    |  +- io.netty:netty-transport:jar:4.1.48.Final:compile
[INFO]    |  \- io.netty:netty-codec:jar:4.1.48.Final:compile
[INFO]    +- io.netty:netty-transport-native-epoll:jar:4.1.48.Final:compile
[INFO]    |  \- io.netty:netty-transport-native-unix-common:jar:4.1.48.Final:compile
[INFO]    +- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO]    +- org.slf4j:slf4j-log4j12:jar:1.7.25:compile
[INFO]    \- log4j:log4j:jar:1.2.17:compile
```

Adding this dependency to a plugin that depends on Jenkins 2.365, one can see the benefit of the managed, provided dependency nicely:

```
[INFO] Assembling webapp text-finder in /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder
[WARNING] Bundling transitive dependency commons-lang-2.6.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-buffer-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-codec-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-common-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-handler-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-resolver-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-native-epoll-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-native-unix-common-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency audience-annotations-0.5.0.jar (via zookeeper)
[WARNING] Bundling transitive dependency zookeeper-jute-3.6.1.jar (via zookeeper)
[INFO] Bundling direct dependency zookeeper-3.6.1.jar
[WARNING] Bundling transitive dependency slf4j-api-1.7.36.jar (via zookeeper)
[WARNING] Bundling transitive dependency slf4j-log4j12-1.7.25.jar (via zookeeper)
[INFO] Generating hpi /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder.hpi
[INFO] Building jar: /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder.hpi
```

No log4j 1.2.17 bundled in the JPI! That is because it is not on the runtime classpath, and Maven HPI plugin only considers this classpath when bundling JARs into JPIs. This is good.

Now suppose we remove the managed, provided dependency on log4j 1.2.17 from the core BOM (as in this PR). I tried this and updated the same plugin to 2.366-SNAPSHOT. Trying to compile the plugin, I now get:

```
[INFO] --- maven-enforcer-plugin:3.1.0:enforce (display-info) @ text-finder ---
[ERROR] Rule 4: org.apache.maven.plugins.enforcer.BannedDependencies failed with message:
Found Banned Dependency: log4j:log4j:jar:1.2.17
Use 'mvn dependency:tree' to locate the source of the banned dependencies.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.715 s
[INFO] Finished at: 2022-08-23T08:36:24-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.1.0:enforce (display-info) on project text-finder: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
```

In other words, the build fails (good!). That Enforcer rule has been in place since plugin POM 2.0; in other words, the vast majority of the plugin ecosystem should have it already in place. To get the build to pass, I would have to do one of three things:

- Exclude log4j from the Zookeeper dependency, which would also prevent it from being bundled in the JPI and picking up SLF4J's version from core at runtime (good!)
- Upgrade Zookeeper to a newer version that doesn't depend on log4j 1.2.17 (good!)
- Disable Enforcer (not ideal)

Disabling Enforcer with `-Denforcer.skip` does in fact allow the problematic JAR to be bundled:

```
[INFO] Assembling webapp text-finder in /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder
[WARNING] Bundling transitive dependency commons-lang-2.6.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-buffer-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-codec-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-common-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-handler-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-resolver-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-native-epoll-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-native-unix-common-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency netty-transport-4.1.48.Final.jar (via zookeeper)
[WARNING] Bundling transitive dependency log4j-1.2.17.jar (via zookeeper)
[WARNING] Bundling transitive dependency audience-annotations-0.5.0.jar (via zookeeper)
[WARNING] Bundling transitive dependency zookeeper-jute-3.6.1.jar (via zookeeper)
[INFO] Bundling direct dependency zookeeper-3.6.1.jar
[WARNING] Bundling transitive dependency slf4j-api-2.0.0.jar (via zookeeper)
[WARNING] Bundling transitive dependency slf4j-log4j12-1.7.25.jar (via zookeeper)
[INFO] Generating hpi /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder.hpi
[INFO] Building jar: /home/basil/src/jenkinsci/text-finder-plugin/target/text-finder.hpi
```

This would presumably trip up scanners, but it would not result in the use of the vulnerable library at runtime, since core's copy of `log4j-over-slf4j` would take precedence.

---

What am I to conclude from all this? Keeping log4j 1.2.17 as a managed, provided dependency is providing very little value. It only exists to serve a narrow use case: to prevent a plugin that consumes log4j 1.2.17 directly or transitively _and_ has explicitly disabled Enforcer from bundling a (vulnerable!) log4j 1.2.17 JAR in its JPI file that would _still_ not be used at runtime.

To me, that is a very niche use case. In other words, I do not think the benefit is worth the cost. So in this PR I am removing the managed, provided dependency on log4j 1.2.17.

### Proposed changelog entries

Developer: Remove managed, provided dependency on log4j 1.2.17. Plugin builds that depend directly or transitively on log4j 1.2.17 must remove this dependency or exclude it from their build in order for their build to pass.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7028"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**
